sdk/trace/provider.go

```diff
@@ -256,11 +253,11 @@ func (p *TracerProvider) Shutdown(ctx context.Context) error {
 		sps.state.Do(func() {
 			err = sps.sp.Shutdown(ctx)
 		})
-		if err != nil {
-			return err
+		if retErr == nil {
+			retErr = err
```

Comment:
This means only the last error will be reported. If several shutdowns failed, we would therefore be missing context.
How about concatenating them instead?

Reply:
Sure. Is there a recommended way to do this?

Reply:
Yes, with `fmt.Errorf` and the `%w` statement, which is treated by Go as an error:

```golang
retErr = fmt.Errorf("%w; %w", retErr, err)
```

Reply:
I'm happy to make this change but semantically that would not be a list of independent errors, it'd be a chain of errors. Are we ok with that? Just want to ensure this is intentional.

Reply:
This is the official recommended way from Go. So yes.
Also, Go does provide an Unwrap method to get each independent error.

Reply:
https://pkg.go.dev/fmt#Errorf says it is invalid to include `%w` more than once and that code [does not work](https://go.dev/play/p/A3IQCrFf1UG) (prints `e1; %!w(*errors.errorString=&{e2})`). I think it's the official way to wrap an error (which is not what we want here), not to "concatenate" two errors.

Reply:
You're right, my bad. We can't change the method signature to return something different than `error`.
So I'd lean towards concatenating all the error strings into a single one, maybe separated by a new line?

Reply:
No worries. I've pushed a slight modification of your idea with semicolons. WDYT?